gensim/models/coherencemodel.py

```diff
@@ -132,6 +143,18 @@ def __init__(self, model=None, topics=None, texts=None, corpus=None, dictionary=
             self.conf = indirect_confirmation_measure.cosine_similarity
             self.aggr = aggregation.arithmetic_mean
 
+        elif self.coherence == 'c_uci':
+            self.seg = segmentation.s_one_one
+            self.prob = probability_estimation.p_boolean_sliding_window
+            self.conf = direct_confirmation_measure.log_ratio_measure
+            self.aggr = aggregation.arithmetic_mean
+
+        elif self.coherence == 'c_npmi':
+            self.seg = segmentation.s_one_one
+            self.prob = probability_estimation.p_boolean_sliding_window
+            self.conf = direct_confirmation_measure.normalized_log_ratio_measure
+            self.aggr = aggregation.arithmetic_mean
+
     def __str__(self):
         return "CoherenceModel(segmentation=%s, probability estimation=%s, confirmation measure=%s, aggregation=%s)" % (
             self.seg, self.prob, self.conf, self.aggr)
```

Comment:
Also how does `_get_topics` below deal with @bhargavvader 's DTM code? He had to make some adjustments to fit it in.